Makefile

```diff
@@ -3,6 +3,7 @@
 	    test unittest integration \
 	    cross localcross
 
+DOCKER := docker
```

Comment:
Can't you call this 
CONTAINER_ENGINE := docker

Reply:
yes, a more generic name would be better

Reply:
CONTAINER_ENGINE seems too long?
CONTAINER?

Reply:
As we already have `DOCKER_RUN_PROXY`, `DOCKER` seems fine.

Reply:
WDYT?

Reply:
Finishing your sentence a bit differently:

> @AkihiroSuda: As we already have `DOCKER_RUN_PROXY`...

... longer names like `CONTAINER_ENGINE` seem fine. ;-)

IMO, `DOCKER` is misleading and `CONTAINER` is too ambiguous. I like `CONTAINER_ENGINE` (but it could easily be another `CONTAINER_xxx` name).

Reply:
updated to use `CONTAINER_ENGINE`